catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SortedQueryMonitor.java

```diff
@@ -148,10 +157,11 @@ public void run() {
                     returnProperties.putAll(properties);
                 }
             } catch (InterruptedException e) {
-                if (source != null) {
+                if (queryRequest != null) {
                     // First, add interrupted processing detail for this source
-                    LOGGER.debug("Search interrupted for {}", source.getId());
-                    processingDetails.add(new ProcessingDetailsImpl(source.getId(), e));
+                    LOGGER.debug("Search interrupted for {}", sourceId);
+                    executePostFederationQueryPlugins(sourceResponse, queryRequest);
```

Comment:
`sourceResponse` will be null at this point. We should be creating a `SourceResponse` with the new `ProcessingDetails` from the next line and sending that in instead.